platforms/software/maven/src/main/java/org/gradle/api/publish/maven/internal/publisher/MavenRemotePublisher.java

```diff
@@ -49,7 +52,8 @@ public MavenRemotePublisher(Factory<File> temporaryDirFactory, BuildCommencedTim
     }
 
     @Override
-    public void publish(MavenNormalizedPublication publication, MavenArtifactRepository artifactRepository) {
+    public void publish(MavenNormalizedPublication publication, @Nullable MavenArtifactRepository artifactRepository) {
```

Comment:
🤔 That's weird. Why did you declare `artifactRepository` as Nullable to have an assert it being not null on the line below?

Reply:
(sorry for the late reply)

Reply:
I had to add the `@NonNullApi` annotation at the class level because one of the CI tests (InternalNullabilityTest?) was failing if I didn't. Since the base `MavenPublisher` interface declares that parameter as `@Nullable`, I had to say it was `@Nullable` in order to make Intellij warning go away. I added the assertion because in this particular implementation of the `MavenPublisher` interface, the parameter cannot be null. Should I remove assert and let it NPE on the next line?

![image](https://github.com/gradle/gradle/assets/1556777/bc854e39-f106-4a9f-8cac-aeb93290c824)

Reply:
A-ha. Indeed, there is a problem with class hierarchy in that area. I don't think this should prevent this PR from merging. That being said, I think it's ok to merge this.

Thanks @hdeadman!